### PR TITLE
Add api changes

### PR DESCRIPTION
**Pull request by D-Triveni**

### CODE REVIEW

File: zephyr/src/indigo_api_callback_dut.c [hunk 1]
```diff
@@ -131,7 +131,7 @@ static int get_control_app_handler(struct packet_wrapper *req, struct packet_wra
 static int reset_device_handler(struct packet_wrapper *req, struct packet_wrapper *resp) {
     int status = TLV_VALUE_STATUS_NOT_OK;
     char *message = TLV_VALUE_RESET_NOT_OK;
-    char role[TLV_VALUE_SIZE], log_level[TLV_VALUE_SIZE], band[TLV_VALUE_SIZE];
+    char role[32], log_level[32], band[32];
```
Comment (krish2718) [inline zephyr/src/indigo_api_callback_dut.c:135]: The moment you write `also` in the commit log that means the change can go in another commit.

File: zephyr/src/utils.c [hunk 2]
```diff
@@ -660,6 +660,20 @@ int find_interface_ip(char *ipaddr, int ipaddr_len, char *name) {
 }
 
 int get_mac_address(char *buffer, int size, char *interface) {
+
+    STRUCT_SECTION_FOREACH(net_if, iface)
+    {
+        const struct device *dev = net_if_get_device(iface);
+        if(!strcmp(dev->name, interface)) {
+            sprintf(buffer, "%02x:%02x:%02x:%02x:%02x:%02x",
+                    iface->if_dev->link_addr.addr[0]&0xff, iface->if_dev->link_addr.addr[1]&0xff,
```
Comment (krish2718) [inline zephyr/src/utils.c]: spaces around `&`.

Comment (krish2718) [inline zephyr/src/utils.c]: Looking at this code its obvious you are access internal structures which shouldn't be done as its buggy and can always break, please use the public API `net_if_get_link_addr` to get `net_linkaddr` and then validate that `net_linkaddr` is actually a valid WI-Fi MAC address and then store to `mac[6]` and print.

File: zephyr/src/utils.c [hunk 3]
```diff
@@ -660,6 +660,20 @@ int find_interface_ip(char *ipaddr, int ipaddr_len, char *name) {
 }
 
 int get_mac_address(char *buffer, int size, char *interface) {
+
+    STRUCT_SECTION_FOREACH(net_if, iface)
```
Comment (krish2718) [inline zephyr/src/utils.c]: Again these are internal details which can easily throw away iterable sections and move to another storage implementation and break this code, please use the public API `net_if_foreach`  + `net_if_is_wifi`.

File: zephyr/src/indigo_api_callback_dut.c [hunk 4]
```diff
@@ -1168,8 +1168,6 @@ static int assign_static_ip_handler(struct packet_wrapper *req, struct packet_wr
     reset_interface_ip(ifname);
     /* Bring up interface */
     control_interface(ifname, "up");
-    /* Set IP address with network mask */
-    strcat(buffer, "/24");
```
Comment (krish2718) [inline zephyr/src/indigo_api_callback_dut.c:1173]: Well, its not supported along with IP, but you can always set mask separately `net_if_ipv4_set_netmask`

File: zephyr/src/utils.c [hunk 5]
```diff
@@ -656,7 +661,22 @@ int send_broadcast_arp(char *target_ip, int *send_count, int rate) {
 }
 
 int find_interface_ip(char *ipaddr, int ipaddr_len, char *name) {
-    return 0;
+    struct wpa_supplicant *wpa_s;
+    char tmp[32];
+
+    if (strcmp(name, get_wireless_interface()))
+        return 0;
+    wpa_s = z_wpas_get_handle_by_ifname(name);
```
Comment (krish2718) [inline zephyr/src/utils.c]: add some newlines to separate logical blocks.

File: zephyr/src/utils.c [hunk 6]
```diff
@@ -656,7 +661,22 @@ int send_broadcast_arp(char *target_ip, int *send_count, int rate) {
...
+    if (!wpa_s) {
+        indigo_logger(LOG_LEVEL_ERROR, "Unable to find the interface: %s, quitting", name);
+        return 0;
+    }
+
+    if (wpa_s->l2 && l2_packet_get_ip_addr(wpa_s->l2, tmp, sizeof(tmp)) >= 0) {
```
Comment (krish2718) [inline zephyr/src/utils.c]: Please note this only gives you IPV4 address

File: zephyr/src/indigo_api_callback_dut.c [hunk 7]
```diff
@@ -1983,7 +1983,7 @@ static int send_sta_disconnect_handler(struct packet_wrapper *req, struct packet
 static int send_sta_reconnect_handler(struct packet_wrapper *req, struct packet_wrapper *resp) {
     struct wpa_ctrl *w = NULL;
     char *message = TLV_VALUE_WPA_S_RECONNECT_NOT_OK;
-    char buffer[256], response[1024];
+    char buffer[256], response[8];
```
Comment (krish2718) [inline zephyr/src/indigo_api_callback_dut.c:1979]: This should be a `fixup! [nrf noup] <original issues>`

File: zephyr/src/utils.c [hunk 8]
```diff
@@ -1344,4 +1344,3 @@ int insert_wpa_network_config(char *config) {
 
 void remove_pac_file(char *path) {
 }
-
```
Comment (krish2718) [inline zephyr/src/utils.c:1307]: should be a fixup of original `no up` commit.

File: zephyr/src/indigo_api_callback_dut.c (hunk 7 above)
Comment (D-Triveni) [reply zephyr/src/indigo_api_callback_dut.c:1979]: > This should be a `fixup! [nrf noup] <original issues>`

Changes are part of the initial commit in which zephyr support was added and the commit got merged. so not using any fixup sauce tag.

File: zephyr/src/utils.c (hunk 8 above)
Comment (D-Triveni) [reply zephyr/src/utils.c:1307]: > should be a fixup of original `no up` commit.
As the original` no up` commit was merged, not using fixup sauce tag for this commit.

**Review by krish2718 — APPROVED** [review]

File: zephyr/src/utils.c (hunk 6 above)
Comment (D-Triveni) [reply zephyr/src/utils.c]: @krish2718   Made the changes to read both ipv4 and ipv6 addresses, kindly review.

File: zephyr/src/utils.c [hunk 9]
```diff
@@ -675,6 +675,33 @@ void set_netmask(char *ifname) {
...
+    char tmp[32];
+
+    if (strcmp(name, get_wireless_interface()))
+        return 0;
+
+    const struct device *dev = device_get_binding(name);
+    struct net_if *wifi_iface = net_if_lookup_by_dev(dev);
+
+    if (net_if_is_wifi(wifi_iface)) {
+#ifdef CONFIG_NET_IPV4
```
Comment (krish2718) [inline zephyr/src/utils.c]: What if both IPV4 and IPV6 are enabled and the interface only has v6 or v4?

File: zephyr/src/utils.c [hunk 10]
```diff
@@ -756,6 +756,27 @@ int control_interface(char *ifname, char *op) {
...
+	indigo_logger(LOG_LEVEL_ERROR, "Cannot find network interface: %s", dev->name);
+	return -1;
+    }
+
+    if (sizeof(ip) > 1) {
+        if (net_addr_pton(AF_INET, ip, &addr)) {
+            indigo_logger(LOG_LEVEL_ERROR, "Invalid address: %s", ip);
+            return -1;
+        }
+        net_if_ipv4_addr_add(wlan_iface, &addr, NET_ADDR_MANUAL, 0);
```
Comment (krish2718) [inline zephyr/src/utils.c:777]: So, while reading we are supporting both IPv4 and IPv6 but while setting only IPv4? Does QT use IPv6 in any of the tests, if not we can just stick to IPv4 to keep things simple.

**Review by bama-nordic — APPROVED** [review]

File: zephyr/src/utils.c (hunk 10 above)
Comment (D-Triveni) [reply zephyr/src/utils.c:777]: Qt is not using Ipv6 in none of the cases. So will remove reading ipv6 changes.

File: zephyr/src/utils.c (hunk 9 above)
Comment (D-Triveni) [reply zephyr/src/utils.c]: Cleaned up reading IPv6 address changes, as QT has only IPv4 support tests.
